scripts/corelli/calibration/bank.py

```diff
@@ -180,10 +186,83 @@ def collect_bank_fit_results(output_workspace: str,
+    tube_count = table.rowCount()  # number of tubes in the bank
+    for tube_index in range(tube_count):
+        positions = np.array(list(table.row(tube_index).values())[1:])  # peak positions for the current tube
+        deviations.append(np.sqrt(np.mean(np.square(positions - positions_average))))
+
+    # find tubes with a large Z-score
+    outlier_values = list()
+    values = deepcopy(deviations)
+    z_score = 1000
+    outlier_value = 1000
+    while z_score > zscore_threshold and outlier_value > deviation_threshold and len(values) > 0:
+        # find the tube with the highest Z-score, possibly signaling a large deviation from the mean
+        mean, std = np.mean(values), np.std(values)
+        outlier_index = np.argmax(np.abs((values - mean) / std))
+        outlier_value = values[outlier_index]
+        # recalculate the Z-score of the tube, but removing it from the pool of values. This removes
+        # any skewing effects from including the aberrant tube in the calculation of its Z-score
+        del values[outlier_index]
+        mean, std = np.mean(values), np.std(values)
+        z_score = np.abs((outlier_value - mean) / std)
+        if z_score > zscore_threshold and outlier_value > deviation_threshold:
+            outlier_values.append(outlier_value)
+
+    # flag the outlier tubes as failing the criterion
+    criterion_pass = np.tile(True, tube_count)  # initialize as all tubes passing the criterion
+    if len(outlier_values) > 0:
+        failure_indexes = [deviations.index(value) for value in outlier_values]
+        criterion_pass[failure_indexes] = False
+
+    # create an analysis summary if so requested
+    if isinstance(summary, str) and len(summary) > 0:
+        success = [1 if criterion else 0 for criterion in criterion_pass]
+        x_values = list(range(1, 1 + TUBES_IN_BANK))
+        mean, std = np.mean(deviations), np.std(deviations)
+        z_scores = np.abs((deviations - mean) / std)
+        y_values = np.array([success, deviations, z_scores]).flatten()
+        workspace = CreateWorkspace(x_values, y_values, NSpec=3, OutputWorkspace=summary,
+                                    WorkspaceTitle='Tube deviations from averages taken over the bank',
+                                    YUnitLabel='Pixel Units', EnableLogging=False)
+        labels = ('success', 'deviation', 'Z-score')
+        axis = TextAxis.create(len(labels))
+        [axis.setLabel(index, label) for index, label in enumerate(labels)]
+        workspace.replaceAxis(1, axis)
+
+    return criterion_pass
+
+
+def criterion_peak_pixel_position(peak_table: InputTable,
                                   summary: Optional[str] = None,
                                   zscore_threshold: float = 2.5,
-                                  deviation_threshold: float = 3.0) -> np.ndarray:
+                                  deviation_threshold: float = 0.0035) -> np.ndarray:
```

Comment:
I'd add a note on the hardcoded value when documenting parameters

Reply:
actually this is wrong. "3" is correct for function `criterion_peak_pixel_position`
I describe now in the docstring that the default of `3` corresponds to the height of three CORELLI pixels